rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -1520,37 +1514,48 @@ bool SPUThread::set_ch_value(u32 ch, u32 value)
 
 	case SPU_WrDec:
 	{
+		bool is_already_occur = ((value & ~(dec_state & dec_run ? (ch_dec_value - (u32)(get_timebased_time() - ch_dec_start_timestamp)) : ch_dec_value)) >> 31) == 1;
 		ch_dec_start_timestamp = get_timebased_time();
 		ch_dec_value = value;
-		return true;
-	}
 
-	case SPU_WrEventMask:
-	{
-		// detect masking events with enabled interrupt status
-		if (value & ~SPU_EVENT_INTR_IMPLEMENTED && interrupts_enabled)
+		if (is_already_occur)
 		{
-			fmt::throw_exception("SPU Interrupts not implemented (mask=0x%x)" HERE, value);
+			dec_state = dec_msb + dec_run;
+			set_events(SPU_EVENT_TM);
 		}
+		else {
+			bool add_thread = (dec_state & dec_upd) == 0;
+			dec_state = (dec_upd + dec_run) | (value >> 31);
 
-		// detect masking unimplemented events
-		if (value & ~SPU_EVENT_IMPLEMENTED)
-		{
-			break;
+			if (add_thread) std::thread (&SPUThread::decrementer_thread ,this).detach();
```

Comment:
I suspect this PR adds a lot of slowdown to games because it can create a lot of very short lived threads. At least this is true for Dante's Inferno where the fps tanks down to ~ 1fps. Since every SPE only has one counter, a thread for each one could be created once and reused. Or maybe just one thread to serve the counter events for all SPEs.

Reply:
seems hard but might worth the effort